hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/entity/EntityRecordItemListener.java

```diff
@@ -352,7 +356,9 @@ private void insertFileData(long consensusTimestamp, byte[] contents, FileID fil
         boolean addressBook = addressBookService.isAddressBook(entityId);
 
         // We always store file data for address books since they're used by the address book service
-        if (addressBook || entityProperties.getPersist().isFiles() ||
+        if (addressBook) {
+            fileDataRepository.save(fileData);
```

Comment:
Can merge the two address book if statements to one:
```suggestion
        if (addressBookService.isAddressBook(entityId)) {
            fileDataRepository.save(fileData);
            addressBookService.update(fileData);
        } else if (entityProperties.getPersist().isFiles() ||
```

Reply:
Merged